Espnet3/installation

Masao-Someki opened the pull request:
## What?

Installation/CI fix for ESPnet-3 development. Work done by @Fhrozen 

## Why?

1. espnet-3 requires additional dependencies.
2. Since espnet installation contains so many dependencies, and some of them are not required for some tasks. We wanted to clean the dependencies based on use cases.

## See also

#6133 


## Major changes

`.devcontainer/`

- Would like to ask Nelson to explain this part, but basically creating devcontainer for ESPnet-3, and refactoring the code base.

`.github/`

- Addition of utilities to simplify environment setup for CI.


`.pre-commit-config.yaml`

- Commented out to temporarily disable functionality for automatic format execution.
- Will uncomment before merge.


`ci/`

- espnet-2: Mainly format and task additions. Want to run CI in parallel by adding tasks as arguments.
- For espnet-1 related items, changed paths to accommodate folder structure changes in test_utils.

- Removed espnet-ez related CI. (ESPnet-3 CI will be added in the coming PR)


`doc`, `vuepress`

- Homepage changes and documentation addition for devcontainer.


`egs*/espnet*`

- I'm sorry, I will delete items that is still not be reverted for formatting.
- For some changes are not related to formats, so currently asking to Nelson.

`test/` 

- Support `torch>2.6`.


`test_utils/`

- Moved espnet1 resources to `espnet1_scripts`.


`tools/` 
- torch2.7 support
- Changes to makefile to use pyproject.toml instead of setup.py.

sw005320 commented:
- Please also change the installation document accordingly. The installation document must be located at https://espnet.github.io/espnet/installation.html
- BTW, https://espnet.github.io/espnet/ does not have a pointer to the installation step, which is the most important documentation.

Masao-Someki commented:
Thank you for your comment!

>BTW, https://espnet.github.io/espnet/ does not have a pointer to the installation step, which is the most important documentation.

Thank you, I didn't realized it...!! I just reached out to @juice500ml for fixing it.

juice500ml commented:
Added requested fixes for docs in https://github.com/espnet/espnet/pull/6135/commits/43b2f1bdde25a2b455edbc117d8c2539ad38cadb

Masao-Someki commented:
@Fhrozen Could you check the modification on the following files are related to installation/CI?

- espnet/nets/pytorch_backend/transformer/embedding.py
- espnet2/asr/encoder/whisper_encoder.py
- espnet2/train/collate_fn.py (I think this is related to CI, just for confirmation)

Masao-Someki commented:
Thank you @juice500ml!
I feel it might be better to replace this block in the former one?
https://github.com/espnet/espnet/blob/43b2f1bdde25a2b455edbc117d8c2539ad38cadb/doc/index.md?plain=1#L18-L21

Fhrozen commented:
> @Fhrozen Could you check the modification on the following files are related to installation/CI?
> 
> * espnet/nets/pytorch_backend/transformer/embedding.py
> * espnet2/asr/encoder/whisper_encoder.py
> * espnet2/train/collate_fn.py (I think this is related to CI, just for confirmation)

Sure np

juice500ml commented:
Discussed w Masao offline, applied his suggestions in d51d90f (change url from tutorial to installation)

sw005320 commented:
> Added requested fixes for docs in [43b2f1b](https://github.com/espnet/espnet/commit/43b2f1bdde25a2b455edbc117d8c2539ad38cadb)
> 
> "Full ESPnet Installation" link added <img alt="image" width="1438" src="https://private-user-images.githubusercontent.com/7028312/453206937-651d8210-503a-4270-8bda-44c65b2a3abe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDk1MDYyODUsIm5iZiI6MTc0OTUwNTk4NSwicGF0aCI6Ii83MDI4MzEyLzQ1MzIwNjkzNy02NTFkODIxMC01MDNhLTQyNzAtOGJkYS00NGM2NWIyYTNhYmUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDlUMjE1MzA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGZlZGU4MjZiYTcyMDM3ZWVkYjE1YzMwYjYzZmM4ODhhOWFiMDAwY2RiYzg2NTlhYzExZjk2Y2RmYjBhZDFkYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.6LsCqbIDeCVmOxw9EXQEIEt9fklIlrwoRknxfH-tXAs">
> 
> New navbar tab "Tutorials" added <img alt="image" width="1439" src="https://private-user-images.githubusercontent.com/7028312/453207059-46bfad42-5c14-4bb7-9b04-1dd7938fa1f2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDk1MDYyODUsIm5iZiI6MTc0OTUwNTk4NSwicGF0aCI6Ii83MDI4MzEyLzQ1MzIwNzA1OS00NmJmYWQ0Mi01YzE0LTRiYjctOWIwNC0xZGQ3OTM4ZmExZjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDlUMjE1MzA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmZmNTYxNmUxMjEzNjhiMGY5Njg2YmEzNGFlNGY5ZTUxZDBiY2M3Njc2Zjc4MmUxZmQ1Njk5YjQzNTk5MWI0OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.hqTPTGvkmhGQ4aFs_N9Dr2AtvguHQcwHYFEI9RREqI4">

Can you add the installation guide to the "Get started with ESPnet!" section or a similar place (you may create a new field)?
The installation part should be located in a visible place. I think an additional item in the "Get started with ESPnet!" section

juice500ml commented:
@sw005320 
The third tab (High performance...) leads to the installation page, per https://github.com/espnet/espnet/commit/d51d90fb04d50a78f45c09790118b2095100f17b . Or maybe it should be more verbose?
<img width="1008" alt="image" src="https://github.com/user-attachments/assets/4910f523-bfca-4aab-a761-8fa5fa67b85e" />

sw005320 commented:
> @sw005320 The third tab (High performance...) leads to the installation page, per [d51d90f]

This is very indirect for me.
Can you clarify more about this description, then?

sw005320 commented:
> I forgot that formatting is automatically done by precommit.ci. I just revert formatting and pushed without triggering any hooks.
> 
> @ftshijt Could you review this PR if it is okay to merge? I will also continue working on adding espnet-3 contents based on this PR!!

You still have more than 200 files.
I believe something is wrong.

Masao-Someki commented:
Oh, it seems precommit document is wrong? I will check more in detail on how to turn it off

sw005320 commented:
> Oh, it seems precommit document is wrong? I will check more in detail on how to turn it off

Formatting issues happen in all Python files under `egs` and `egs2`

Masao-Someki commented:
@ftshijt @sw005320 @Fhrozen
Sorry for the formatting issues earlier, and thank you for your help, @Fhrozen.
There are now 85 files that require review. Could you please take a look at the PR?

Fhrozen commented on test/espnet2/bin/test_cls_inference.py:
```diff
@@ -12,6 +12,8 @@
 from espnet2.bin.cls_inference import Classification, get_parser, inference, main
 from espnet2.tasks.cls import CLSTask
 
+# import yaml
```
probably, we can remove it

Fhrozen commented on .github/workflows/ci_on_ubuntu.yml:
```diff
@@ -252,40 +264,87 @@ jobs:
-
+          flags: test_configuration_espnet2_${{ matrix.task }}
 
   test_import:
```
This test is making CI fail. Can you comment it? I will need to modify it later. 
Especially, with the new changes for ESPnet2 installation, it requires moving some files between directories during the installation process.  I will be activating it later

Fhrozen commented:
> @ftshijt @sw005320 @Fhrozen Sorry for the formatting issues earlier, and thank you for your help, @Fhrozen. There are now 85 files that require review. Could you please take a look at the PR?

There are some files that require formatting and deleting some unused packages.

Masao-Someki commented:
@sw005320 @ftshijt 
Here's the summary on this PR.

@Fhrozen Could you please add a description of the changes in `.devcontainer` if the reviewer asks about it? Also, if any of my descriptions below are incorrect, please let me know!

## Major changes

`.devcontainer/`

- Would like to ask Nelson to explain this part, but basically creating devcontainer for ESPnet-3, and refactoring the code base.

`.github/`

- Addition of utilities to simplify environment setup for CI.


`.pre-commit-config.yaml`

- Commented out to temporarily disable functionality for automatic format execution.
- Will uncomment before merge.


`ci/`

- espnet-2: Mainly format and task additions. Want to run CI in parallel by adding tasks as arguments.
- For espnet-1 related items, changed paths to accommodate folder structure changes in test_utils.

- Removed espnet-ez related CI. (ESPnet-3 CI will be added in the coming PR)


`doc`, `vuepress`

- Homepage changes and documentation addition for devcontainer.


`egs*/espnet*`

- I'm sorry, I will delete items that is still not be reverted for formatting.
- For some changes are not related to formats, so currently asking to Nelson.

`test/` 

- Support `torch>2.6`.


`test_utils/`

- Moved espnet1 resources to `espnet1_scripts`.


`tools/` 
- torch2.7 support
- Changes to makefile to use pyproject.toml instead of setup.py.

---

Thank you very much for all your work on this. I would greatly appreciate your review!

sw005320 commented:
> @sw005320 @ftshijt Here's the summary on this PR.
> 
> @Fhrozen Could you please add a description of the changes in `.devcontainer` if the reviewer asks about it? Also, if any of my descriptions below are incorrect, please let me know!
> 
> ## Major changes
> `.devcontainer/`
> 
> * Would like to ask Nelson to explain this part, but basically creating devcontainer for ESPnet-3, and refactoring the code base.
> 
> `.github/`
> 
> * Addition of utilities to simplify environment setup for CI.
> 
> `.pre-commit-config.yaml`
> 
> * Commented out to temporarily disable functionality for automatic format execution.
> * Will uncomment before merge.
> 
> `ci/`
> 
> * espnet-2: Mainly format and task additions. Want to run CI in parallel by adding tasks as arguments.
> * For espnet-1 related items, changed paths to accommodate folder structure changes in test_utils.
> * Removed espnet-ez related CI. (ESPnet-3 CI will be added in the coming PR)
> 
> `doc`, `vuepress`
> 
> * Homepage changes and documentation addition for devcontainer.
> 
> `egs*/espnet*`
> 
> * I'm sorry, I will delete items that is still not be reverted for formatting.
> * For some changes are not related to formats, so currently asking to Nelson.
> 
> `test/`
> 
> * Support `torch>2.6`.
> 
> `test_utils/`
> 
> * Moved espnet1 resources to `espnet1_scripts`.
> 
> `tools/`
> 
> * torch2.7 support
> * Changes to makefile to use pyproject.toml instead of setup.py.
> 
> Thank you very much for all your work on this. I would greatly appreciate your review!

You can move this to the top message of this PR.

Masao-Someki commented:
Thank you, I added the major changes section at the initial comment!

Masao-Someki commented:
@Fhrozen Could you check the CI issue..? If my understanding is correct, those errors basically comes from the formatting?

For the issue in `generate_and_deploy_doc`, the reason is the lack of dependencies in `pyproject.toml` for docs. I asked @juice500ml to fix it.

ftshijt commented:
One proposal: maybe we can remove the espnet1 support from espnet3 by removing the related CI tests? It has been 6 years since the espnet2 cames out, we might need support window to maximize our supporting efforts

juice500ml commented:
There were some old configs + missing deps in new `pyproject.toml`, so that `generate_and_deploy_doc` CI were broken. Hopefully 6491fc0 fixes this? (BTW, the new `pyproject.toml` looks great! 👍)

sw005320 commented:
Do you know what's happening?
It looks like this would be another formatting issue.
https://github.com/espnet/espnet/actions/runs/15568526717/job/43838458450?pr=6135

sw005320 commented:
OK, but honestly, I cannot understand it.
Why does this PR break the CI? This PR is not related to them.

Can you use git commands to make some of the files the same as the main repo?

sw005320 commented:
E.g., 
```
git checkout main -- path/to/dir1 path/to/dir2
```

juice500ml commented:
Introducing Tutorials had problems while generating sidebars in documents. 9fb310f and 3a127bc fixes it. Hopefully the document is generated correctly now.

Masao-Someki commented:
I checked the "Files changed" tab, and this PR doesn't modify the files causing the CI failure. I also ran the git command locally and didn’t find any differences.

sw005320 commented on espnet2/asr/encoder/whisper_encoder.py:
Why does this change happen?
This must not be related to this PR.

sw005320 commented:
Is it possible to create a new PR?
This PR has too many irrelevant files, and it is very difficult to proceed with it.

sw005320 commented on espnet2/train/collate_fn.py:
Why does this change happen?
This must not be related to this PR.

Masao-Someki commented:
It's possible to create a new PR, but we need to clearly specify what should be included..

For those particular files, I'm currently asking @Fhrozen to confirm whether the changes are necessary ([link to comment](https://github.com/espnet/espnet/pull/6135#issuecomment-2957095417)).

Aside from those files, I believe the other changes should be included..

sw005320 commented:
I found that 

> `test_utils/`
>
> Moved espnet1 resources to `espnet1_scripts`

has a major issue.
Some scripts are not related to espnet1.
We should check which one we can move to `espnet1_scripts` one by one.

sw005320 commented:
@Fhrozen, is it possible to separate this PR and `test_utils/`? 
`test_utils/` needs some considerations and we may do it later.

Fhrozen commented:
@Masao-Someki @sw005320, The CI is fixed. Let me know if there is any comment or modification is required. I will be moving to another PR as soon as this is merged.

sw005320 commented:
@Fhrozen, can you remove `test_utils/` related from this PR?
This PR is too large and too complex as a single PR.

Masao-Someki commented:
@Fhrozen I can remove the test_utils/ related work, and could you confirm the following process looks okay to you:

1. revert the following commits:
    - [update bats for espnet1](https://github.com/espnet/espnet/commit/ac2b0eaac9b8418e435616ddc5db4d32824cef5e)
    - [rename folder](https://github.com/espnet/espnet/commit/8bfdac64fa2b37e304aed5ad1b26b72fecda3274)
2. Revert the ci fix ([updates for ci](https://github.com/espnet/espnet/commit/c911767789a99126972b179483f1853a53a5c120))

Since the second one contains formatting and weird fix for whisper encoder, I think we can revert all these three commits and can remove `test_utils/espnet1_script` related works.

Fhrozen commented:
@Masao-Someki, just delete the the `espnet1_scripts/` from the files `test_utils/integration_test_translate_wav.bats`
and `test_utils/integration_test_asr_align_wav.bats`
